openapi.yaml

```diff
@@ -25,254 +37,677 @@ security:
+        - in: path
+          name: doughID
+          required: true
+          type: string
+          description: ID of the dough recipe to retrieve pizzas for
+        - in: body
+          name: filters
+          description: "Attributes to restrict pizza results to"
+          schema:
+            $ref: "#/definitions/PizzaAttributes"
+      responses:
+        200:
+          description: "A list of pizzas"
+          schema:
+            $ref: "#/definitions/PizzasResults"
+        404:
+          $ref: "#/responses/404"
+        500:
+          $ref: "#/responses/500"
+  /ingredients:
+    get:
+      description: "Get the a list of all ingredients"
+      tags:
+        - ingredient
+      parameters:
+        - in: query
+          name: name
+          description: Search ingredients by name
+          required: false
+          type: string
+        - in: query
+          name: type
+          description: Search ingredients by type
+          required: false
+          type: string
+      responses:
+        200:
+          description: "A list of ingredients"
+          schema:
+            $ref: "#/definitions/IngredientsResult"
+    post:
+      description: "Create a new ingredient"
+      tags:
+        - ingredient
+      parameters:
+        - in: body
+          name: attributes
+          description: "Attributes to assign to the new ingredient"
+          schema:
+            $ref: "#/definitions/IngredientAttributes"
+      responses:
+        200:
+          description: "The newly created ingredient"
+          schema:
+            $ref: "#/definitions/IngredientResult"
+        500:
+          $ref: "#/responses/500"
+  /ingredients/{ingredientID}:
+    get:
+      description: "Get the ingredient with ID ingredientID"
+      tags:
+        - ingredient
+      parameters:
+        - in: path
+          name: ingredientID
+          required: true
+          type: string
+          description: ID of the ingredient to retrieve
+      responses:
+        200:
+          description: "A single ingredient"
+          schema:
+            $ref: "#/definitions/IngredientResult"
+        404:
+          $ref: "#/responses/404"
+        500:
+          $ref: "#/responses/500"
+    patch:
+      description: "Update the ingredient with ID ingredientID"
+      tags:
+        - ingredient
+      parameters:
+        - in: path
+          name: ingredientID
+          required: true
+          type: string
+          description: ID of the ingredient to retrieve
+        - in: body
+          name: attributes
+          description: attributes to be updated
+          schema:
+            $ref: "#/definitions/IngredientAttributes"
+      responses:
+        200:
+          description: "The updated ingredient"
+          schema:
+            $ref: "#/definitions/IngredientResult"
+        404:
+          $ref: "#/responses/404"
+        500:
+          $ref: "#/responses/500"
+  /ingredients/{ingredientID}/pizzas:
```

Comment:
How if we want to sure all pizzas made with more than one ingredients? For example, how to search all pizzas contain `pineapple` and `mango` (😨) ?

Reply:
I would hope no one's searching for `pineapple` and `mango` pizzas, but if they were we could do the same thing you suggested with the `doughId` and move it to a parameter that would allow you to pass an array of ingredients